model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,437 @@
+      {
+        std::cout << "    - Joint type:  " << jointTypes[jointComp->Data()]
+        << "\n"   << "    - Parent Link: [" << childLinkComp->Data() << "]\n"
+                  << "    - Child Link:  [" << parentLinkComp->Data() << "]\n";
+      }
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+void printBoundingBox(uint64_t entity,
+                ignition::gazebo::EntityComponentManager &ecm)
+{
+  auto bounding_boxes = ecm.EntitiesByComponents(
+  ignition::gazebo::components::ParentEntity(entity),
+  ignition::gazebo::components::AxisAlignedBox());
+
+  for(auto &_entity : bounding_boxes)
+  {
+    {
+      auto parentComp =
+          ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+      auto nameComp =
+          ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+      std::string parentInfo;
+      if (parentComp)
+      {
+        auto parentNameComp =
+            ecm.Component<ignition::gazebo::components::Name>(
+            parentComp->Data());
+
+        if (parentNameComp)
+        {
+          parentInfo += parentNameComp->Data() + " ";
+        }
+        parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+      }
+
+      std::cout << "  - Bounding Box [" << _entity << "]" << std::endl
+                << "    - Name: " << nameComp->Data() << std::endl
+                << "    - Parent: " << parentInfo << std::endl;
+    }
+    {
+      auto boxComp =
+          ecm.Component<ignition::gazebo::components::AxisAlignedBox>(_entity);
+
+      if (boxComp)
+      {
+
+        std::cout << "     Size:   [" << boxComp->Data().Size() << "]\n"
+                  << "     Center: [" << boxComp->Data().Center() << "]\n"
+                  << "     Min:    [" << boxComp->Data().Min() << "]\n"
+                  << "     Max:    [" << boxComp->Data().Max() << "]\n";
+      }
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+extern "C" IGNITION_GAZEBO_VISIBLE void cmdModelInfo(
+  const char *_model, int _pose, int _link, const char *_link_name,
+  int _joint, const char *_joint_name)
+{
+  std::string link_name{""};
+  if(_link_name)
+    link_name = _link_name;
+  std::string joint_name{""};
+  if(_joint_name)
+    joint_name = _joint_name;
+
+  bool printAll{false};
+  if(!_pose && !_link && !_joint)
+    printAll = true;
+
+  std::string model{_model};
+  // Get arguments
+  std::string world = getWorldName();
+  if (world.empty())
+    return;
+  // Create a transport node.
+  ignition::transport::Node node;
+
+  bool executed{false};
+  bool result{false};
+  unsigned int timeout{5000};
+  std::string service{"/world/" + world + "/state"};
+
+  std::cout << std::endl << "Requesting state for world [" << world
+            << "] on service [" << service << "]..." << std::endl << std::endl;
+
+  // Request and block
+  ignition::msgs::SerializedStepMap res;
+  executed = node.Request(service, timeout, res, result);
```

Comment:
Given that how you get a snap of the ecm is identical for this method an for the `cmdModeList` method probably you could define a method that returns an ecm object out of the running simulation and use it here and in the previously mentioned method.

Reply:
Addressed